### ISSUE TITLE
[Python] Feval doesn't work anymore with Maximise v 0.6 

### ISSUE THREAD

**Issue by armgilles** [dmlc/xgboost#1437]
I use a customized evaluation function to use [MAPE](https://en.wikipedia.org/wiki/Mean_absolute_percentage_error) metric and try to minimise it. Actually it worked on v 0.4 (pip version). But the 0.6 version, it appears that feval always want to maximise it.

- **Replicable Code :**

```
import xgboost as xgb
import numpy as np
from sklearn.cross_validation import train_test_split
from sklearn.datasets import load_boston


def mean_absolute_percentage_error(y_true, y_pred):
    """
    MAPE Metric
    """
    return np.mean(np.abs((y_true - y_pred) / y_true)) * 100
    
def mean_absolute_percentage_error_xg(yhat, y):
    """
    Custum metric for Xgboost evaluation
    """
    y = y.get_label()
    yhat = yhat
    return "mape", mean_absolute_percentage_error(y,yhat)
    
print xgb.__version__

boston = load_boston()
y = boston['target']
X = boston['data']

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=2016)


dtrain = xgb.DMatrix(X_train, y_train)
dtest = xgb.DMatrix(X_train, y_train)

evallist = [(dtrain, 'train'), (dtest, 'test')]

params = {'max_depth':3,
         'eta':0.5,
         'objective':'reg:linear',
         'seed':0}
         
xgb_model = xgb.train(params, dtrain, 90, evallist, early_stopping_rounds=10, 
                      maximize=False, feval=mean_absolute_percentage_error_xg)
```

- **Output :**

```
0.6
[0]     train-mape:48.1152      test-mape:48.1152
Multiple eval metrics have been passed: 'test-mape' will be used for early stopping.

Will train until test-mape hasn't improved in 10 rounds.
[1]     train-mape:24.4804      test-mape:24.4804
[2]     train-mape:13.7657      test-mape:13.7657
[3]     train-mape:10.5038      test-mape:10.5038
[4]     train-mape:9.42211      test-mape:9.42211
[5]     train-mape:9.1674       test-mape:9.1674
[6]     train-mape:8.60307      test-mape:8.60307
[7]     train-mape:8.13896      test-mape:8.13896
[8]     train-mape:7.71197      test-mape:7.71197
[9]     train-mape:7.46611      test-mape:7.46611
[10]    train-mape:7.01876      test-mape:7.01876
Stopping. Best iteration:
[0]     train-mape:48.1152      test-mape:48.1152
```

I checked change log but didn't see anything. Did I forget something ?

**Comment by khotilov** [comment]
Simply rename it to MAPE, or anything that doesn't start specifically with map.
See these lines: https://github.com/dmlc/xgboost/blob/7addebb2eac3706f289d40eeeae3762574ea2dfb/python-package/xgboost/callback.py#L161-L166

**Comment by armgilles** [comment]
Yes change ```mape``` to ```MAPE``` works : 
```
def mean_absolute_percentage_error_xg(yhat, y):
    """
    Custum metric for Xgboost evaluation
    """
    y = y.get_label()
    yhat = yhat
    return "MAPE", mean_absolute_percentage_error(y,yhat) # Change here !
```

Thanks @khotilov 👍
